### PR TITLE
Garnettanner/vm setup

### PR DESCRIPTION
**Pull request by Gtanner00**
I've made changes based on Enoch's recommendations. 
The guide now only concerns setting up the virtual machine.  I've also moved it into the general Documentation directory.

Enoch, do you plan to push your changes to the install.sh?
I can improve the guide on how to set up the build environment.

### CODE REVIEW

File: Documentation/LinuxVMGuide/README.adoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
+﻿= Setting up a Linux Virtual Machine
+
+This guide assumes the host operating system is 64 bit Windows
+This guide was written for Linux Mint 18.3.
+The steps include installing the required software on the host machine and setting up Linux Mint as a virtual machine (VM)
+
+== 1. Install Virtualbox
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:7]: Pretty odd to have the numbering here. I dont' think the numbers are necessary.

File: Documentation/LinuxVMGuide/README.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
+﻿= Setting up a Linux Virtual Machine
+
+This guide assumes the host operating system is 64 bit Windows
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:3]: Missing period.

File: Documentation/LinuxVMGuide/README.adoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+This guide was written for Linux Mint 18.3.
+The steps include installing the required software on the host machine and setting up Linux Mint as a virtual machine (VM)
+
+== 1. Install Virtualbox
+Download https://www.virtualbox.org/wiki/Downloads[Oracle Virtual Box]
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:8]: Missing period.

File: Documentation/LinuxVMGuide/README.adoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+The latest version at writting was 5.2.4.
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:9]: Writting -> writing

File: Documentation/LinuxVMGuide/README.adoc [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+The steps include installing the required software on the host machine and setting up Linux Mint as a virtual machine (VM)
+
+== 1. Install Virtualbox
+Download https://www.virtualbox.org/wiki/Downloads[Oracle Virtual Box]
+The latest version at writting was 5.2.4. 
+Download the VirtualBox Extension Pack from the same page.
+Install both.
+
+== 2. Setup a Virtual Machine
+Create a new virtual machine in Virtualbox using the _New_ icon.
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:14]: Headers should have an empty line above and below them.

File: Documentation/LinuxVMGuide/README.adoc [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+
+image::https://github.com/StudentOrganisationForAerospaceResearch/DevelopmentResources/blob/garnettanner/stm32-example/STM32Resources/BuildEnvironment_Setup/images/1.PNG[]
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:17]: These links are very likely to break, you should use relative paths.

You can reference the README from this project.
https://raw.githubusercontent.com/JeLLyNinjas/TerminalFighter/master/README.adoc
https://github.com/JeLLyNinjas/TerminalFighter/blob/master/README.adoc

Usually, if your name appears anywhere in a file, it's a red flag :)

File: Documentation/LinuxVMGuide/README.adoc [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+Choose Linux as the type and Ubuntu (64 bit) as the OS version as Mint is based on Ubuntu.
+Mint reccomends 20 GB of hard disk and 1 GB of RAM for a bare minimal installation. Give yourself extra RAM and space for project files and application packages.
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:20]: One sentence per line

File: Documentation/LinuxVMGuide/README.adoc [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+== 2. Setup a Virtual Machine
+Create a new virtual machine in Virtualbox using the _New_ icon.
+
+
+image::https://github.com/StudentOrganisationForAerospaceResearch/DevelopmentResources/blob/garnettanner/stm32-example/STM32Resources/BuildEnvironment_Setup/images/1.PNG[]
+
+Choose Linux as the type and Ubuntu (64 bit) as the OS version as Mint is based on Ubuntu.
+Mint reccomends 20 GB of hard disk and 1 GB of RAM for a bare minimal installation. Give yourself extra RAM and space for project files and application packages. 
+You can change these settings later if required. 
+Use the default options if unsure
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:22]: I would reword to

`The defaulting are usually fine.`

File: Documentation/LinuxVMGuide/README.adoc [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+=== Install Linux Mint
+Download the https://linuxmint.com/edition.php?id=246[Linux Mint Cinnamon 64 bit ISO].
+When Virtualbox asks to select a start up disk, navigate to the ISO which will boot into the installer.
+Click the disk icon to start the installation and run through it. 
+Select the default options if unsure of anything.
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:28]: I don't think this line is necessary, this was stated before.

File: Documentation/LinuxVMGuide/README.adoc [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+You should now be able to use shared clipboard between the host and guest operating systems once you enable these features in the Devices menu.
+
+Note: Drag and Drop does not work between a Windows Host and Linux VM.
+
+Shut down your VM. Go to Virtualbox Settings->Display->Enable 3D acceleration (if available) and USB-> Enable USB controller. 
+When you plug in a USB device, you can capture it in the VM by clicking on the USB icon in the bottom right of the Virtualbox window and selecting your device.
+You can also add a filter to auto capture certain devices when they are plugged in.
+
+
+==== Extra Tip
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:50]: I would rename this to `Shared Folder between the Host and VM (Optional)`

File: Documentation/LinuxVMGuide/README.adoc [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+Setup a shared folder between Windows and Linux in the Virtualbox Settings->Shared Folders -> Add new Shared Folder.
+Check _Automount_ and _Make Perminant_ After you mount the drive you need to run
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:52]: _Make Perminant_ should be _Make Permanent_ right?

File: Documentation/LinuxVMGuide/README.adoc [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+the following command to give Linux permissions to access the files.
```
Comment (ExtraGravity) [inline Documentation/LinuxVMGuide/README.adoc:53]: This should stick with the previous line, one sentence one line.

**Review by ExtraGravity — CHANGES_REQUESTED** [review]

**Comment by ExtraGravity** [comment]
Don't think garnet is gonna get around to this, which is fine.

**Comment by Gtanner00** [comment]
Sorry this was left unfinished. My time was better served getting hardware ready.
